dali/pipeline/operators/decoder/host_decoder_test.cc

```diff
@@ -20,7 +20,7 @@ template <typename ImgType>
 class HostDecodeTest : public GenericDecoderTest<ImgType> {
  protected:
   uint32_t GetImageLoadingFlags() const override {
-    return t_loadJPEGs + t_loadPNGs;
+    return t_loadJPEGs + t_loadPNGs + t_loadTiffs;
```

Comment:
I am not sure that `t_loadTiffs` or even `t_loadPNGs` should be added here unless such files are really used in the SAME test. Yes, perhaps it was me who started to use `t_loadPNGs` here, but when I did that I have not worsened the test performance. Now the test performance will be worse, because for each test (`TestJPEGDecode`, `TestPNGDecode` or `TestPNGDecode`) we will load files from all THREE group, but we will use ONLY one of group of the files during the test.

The right solution of this problem is a little bit tricky. The fact is: virtual method `GetImageLoadingFlags()` is called in
`DALISingleOpTest::SetUp()` which is always called in test constructor. Therefore, the easiest way to solve it would be the implementation of a new class (for each new type of images) with just one virtual function in it. For instance, for PNG it would be
```
template <typename ImgType>
class HostDecodeTestPNG : public HostDecodeTest<ImgType> {
  uint32_t GetImageLoadingFlags() const override {
    return t_loadPNGs;
  }
};
```
Definitely, the the test itself should be rewritten with that new class:
```
TYPED_TEST_CASE(HostDecodeTestPNG, Types);

TYPED_TEST(HostDecodeTestPNG, TestPNGDecode) {
  this->RunTestDecode(t_pngImgType);
}
```

We will also should remove `GetImageLoadingFlags()` from HostDecodeTest. In that case for JPEGs the program will use
```
  virtual uint32_t DALISingleOpTest::GetImageLoadingFlags() const   {
    return t_loadJPEGs;   // Only loading of JPEG files
  }
```